### DISCUSSION TITLE
Some question about DerivativeMultiPhaseMaterial.i

### DISCUSSION THREAD

**Discussion by AlexSacMorane** [idaholab/moose#18870] [Q&A Modules: Phase field] [answered]
Dear all,

I am studying the example input at **moose/modules/phase_field/examples/multiphase/DerivativeMultiPhaseMaterial.i** to understand how a multi phase simulation works. And I have some questions about it :

- This example uses 3 unconserved variables _etai_ but what should I change if I would like to use conserved variables ? If I understand well [Basic Phase Field Equations](https://mooseframework.inl.gov/modules/phase_field/Phase_Field_Equations.html#!), I just need to replace _ACMultiInterface_ by _CHInterface_ and _AllenCahn_ by _CahnHilliard_.
- Does the free energy used here depend only on the concentration _c_ ? I do not see any _etai_ in the definition of the _phase_free_energy_i_ .

I stay available. Regards

Alexandre Sac--Morane

**Comment by souravmat-git** [comment, marked as answer]
Hello,

I am not sure about the model you are trying to use. 

To my understanding, the ACMultiInterface kernel is based on the model by Steinbach et al. (see Eq. 6 in https://www.sciencedirect.com/science/article/pii/0167278995002987), while CHInterface kernel is needed to implement composition gradients, which is not assumed in this example model.

No. The _total_ free energy F (see https://mooseframework.inl.gov/modules/phase_field/MultiPhase/WBM.html) depends on both concentration and phase-field variable etai. This expression is coded in the material DerivativeMultiPhaseMaterial.

Hope this helps.
Sourav

**Reply by AlexSacMorane** [reply]
Thanks you, I have well noticed the influence of c and etai on the total free energy F. 

Concerning my model, here the simuation considers unconserved variables etai. But I would like to do the same with conserved variables ci. I have just replaced Allen-Cahn equations by Cahn-Hilliard ones. Then I have trouble for the moment with SwitchingFunctionMaterial. I would like to impose the sum of my ci equals to 1 but the SwitchingFunctionMaterial seems to be designed for etai. Is there a same function concerning conserved variables?

Regards
Alexandre

**Reply by anilkunwar** [reply]
Switching functions (function of eta_i) are principally designed to interpolate properties at the interface , and thus two different properties can be assigned to two different phases (eta1 and eta2). Generally, for a system with non-conserved volume, the phases (eta_i) are described by Allen-Cahn equation and conserved variables ci (e.g. composition) are described by Cahn-HIlliard equation. There is no need to define switching function of ci variables (if they mean mole fraction and do not mean phases ). I assume that the system is binary, ternary or multicomponent  (with the statement sum of ci equals 1) . In this case, the design rule for interface composition can be understood through WBM and KKS models (these models are implemented in moose).  In case like non-reactive wetting , volume is conserved and  in such cases Cahn-Hilliard equation represents the evolution of phases (eta or c  whichever notation is used) .

**Reply by AlexSacMorane** [reply]
Ok thanks for those details it becomes clearer. I think I have managed to get what I would like.
